### DISCUSSION TITLE
Connecting TablePlus / Navicat

### DISCUSSION THREAD

**Discussion by yankeeinlondon** [supabase/supabase#552] [Questions] [answered]
Based on the connection information I see in the ` Settings > Database dialog ` I was expecting to be able to connect with tools like TablePlus and/or Navicat to the DB but haven't quite figured out how one does this. Can you confirm that this _should_ be possible? Any tips on getting the network settings right?

![image](https://user-images.githubusercontent.com/980757/104851114-d8804600-58a7-11eb-8071-ea1b522f87cc.png)

>  Probably worth saying that my SQL muscles are very out of shape and my understanding of Postgres specifics are pretty much what I can get from google searches ;)

**Comment by silentworks** [comment, marked as answer]
Yes, this should work with any of those clients. I have tested it with TablePlus and was able to connect to the database.
![Screenshot 2021-01-17 at 18 01 46](https://user-images.githubusercontent.com/79497/104851632-215aed80-58ee-11eb-9029-9ddf76dd50ea.png)

**Reply by yankeeinlondon** [reply]
I had made the mistake of using the Project name as the database name. Doh! I tried with the generic "postgres" and it now works.

**Comment by JamieGoodson** [comment]
TablePlus gives me an error when trying to connect to Supabase using the provided direct connection hostname:

<img width="248" alt="Screenshot 2025-06-16 at 19 54 48" src="https://github.com/user-attachments/assets/1f092f48-539d-45f9-813d-80f8ecc674d3" />


If I try to `ping <my-db-url>` I get `Unknown host`. I can access the database via the supabase.com dashboard just fine.


Edit: The solution was to use the Supabase "Session pooler" connection details rather than "Direct connection". The connection string should look like this:

```
postgresql://postgres.[YOUR-DB-ID]:[YOUR-PASSWORD]@aws-0-us-east-1.pooler.supabase.com:5432/postgres
```

When creating a new connection in TablePlus, select "Import from URL" and paste your session pooler connection string.
<img width="341" alt="image" src="https://github.com/user-attachments/assets/cd9f9aaf-c6b0-47d2-8d73-e2201adc0cbc" />

**Reply by mnah05** [reply]
use the session pooler string that one worked for me!
<img width="998" height="491" alt="Screenshot 2025-12-23 at 3 21 20 PM" src="https://github.com/user-attachments/assets/28f0fc01-4549-49b5-bec9-f43d2926e132" />
